src/main/java/org/jolokia/docker/maven/StartMojo.java

```diff
@@ -145,6 +147,39 @@ private void waitIfRequested(DockerAccess docker, ImageConfiguration imageConfig
             checkers.add(getLogWaitChecker(wait.getLog(), docker, containerId));
             logOut.add("on log out '" + wait.getLog() + "'");
         }
+        if (wait.getTcp() != null) {
+            WaitConfiguration.TcpConfiguration tcpConfig = wait.getTcp();
+            try {
+                Container container = docker.inspectContainer(containerId);
+                String host = tcpConfig.getHost();
+                List<Integer> ports = new ArrayList<>();
+
+                if (host == null) {
+                    // Host defaults to ${docker.host.address}.
+                    host = projectProperties.getProperty("docker.host.address");
+                }
+
+                if ("localhost".equals(host)) {
+                    host = container.getIPAddress();
```

Comment:
Host can be null (e.g. when no ports are exposed). Could this be an issue here ?